pkg/iac/terraform/block.go

```diff
@@ -576,3 +596,122 @@ func (b *Block) IsNil() bool {
 func (b *Block) IsNotNil() bool {
 	return !b.IsNil()
 }
+
+func (b *Block) ExpandBlock() error {
+	var (
+		expanded []*Block
+		errs     []error
+	)
+
+	for _, child := range b.childBlocks {
+		if child.Type() == "dynamic" {
+			blocks, err := child.expandDynamic()
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			expanded = append(expanded, blocks...)
+		}
+	}
+
+	for _, block := range expanded {
+		b.injectBlock(block)
+	}
+
+	return errors.Join(errs...)
+}
+
+func (b *Block) expandDynamic() ([]*Block, error) {
+	if b.IsExpanded() || b.Type() != "dynamic" {
+		return nil, nil
+	}
+
+	realBlockType := b.TypeLabel()
+	if realBlockType == "" {
+		return nil, errors.New("dynamic block must have 1 label")
```

Comment:
we seem to have been using `fmt.Errorf` elsewhere, should we stick to it?

Reply:
`fmt.Errorf` is used to format the string

Reply:
ah doh, didn't even realize that 😆